FrontEnd/src/hooks/useProvideAuth.js

```diff
@@ -4,16 +4,19 @@ import axios from 'axios';
 export default function useProvideAuth() {
   const [isAuth, setIsAuth] = useState(false);
   const [isLoading, setLoading] = useState(true);
+  const [UserId, setUserId] = useState(null);
   const validateToken = async (authToken) => {
     try {
-      await axios.get(
+      const userData = await axios.get(
         `${process.env.REACT_APP_BASE_API_URL}/api/auth/users/me/`,
         {
           headers: {
             Authorization: `Token ${authToken}`,
           },
         }
-      );
+      )
+      .then((res) => res.data);
+      setUserId(userData.id);
       return true;
```

Comment:
You should use existing useUser hook instead of adding userData here.

Reply:
Do you think, we need additional request to api/users/me?

Reply:
useUser hook provides all the information about authorized user. And you can access this data when you need it. 
It is common practice to use this hook. And it's no need to duplicate the logic of getting user id.
And we get that information via request to api/users/me.

Reply:
When I as authorized user put smth to the searchpage and press enter iuseProvideAuth sends the request to api/users/me. So I use it. If I'll use useUser it will send this request again.

Reply:
So we have duplicated logic in our code)

Reply:
Propositions?

Reply:
I'll change code, but I'm not sure in this solution

Reply:
We have useUser hook. My proposition is to use it instead of adding userData in useProvideAuth.

Reply:
I agree with @Lvyshnevska , inside a `useUser` hook we utilize a `useSWR` which should cache fetched data and prevent second API call. There is no need to call `/api/auth/users/me/` the second time if it was already called and cached in the `useUser`

Reply:
Ok. Done

Reply:
As far as what I see we return different values from these hooks. But since they use the same API I assume that we can combine them both into one hook. Also, we do not use SWR in `useProvideAuth` that's something that could be fixed